relay-server/src/actors/project_cache.rs

```diff
@@ -165,6 +165,17 @@ impl UpdateRateLimits {
     }
 }
 
+pub struct BufferIndex {
+    project_key: ProjectKey,
+    keys: BTreeSet<QueueKey>,
+}
+
+impl BufferIndex {
+    pub fn new(project_key: ProjectKey, keys: BTreeSet<QueueKey>) -> Self {
+        Self { project_key, keys }
+    }
+}
```

Comment:
I dislike the complexity of adding this new message type, especially because it now requires passing the buffering service now suddenly needs to have a notion of a `project_key`. Can we make `DequeueMany` have an `AsyncResponse` instead, and we send the `back_keys` via that channel instead?

Reply:
If the `AsyncResponse` is introduced we will have to `await` on it and block, which we want to avoid in the processing pipeline. This will also will require introducing `async`  context where is currently none, project cache and project broker currently sync and do not await on anything. 

The point here was to have everything as async as possible and do not block on any stages.

Reply:
> we will have to await on it and block, which we want to avoid in the processing pipeline.

OK good point. I wonder if the index that maps from project key to queue keys should just move into the buffer service, that would also make this additional message type unnecessary (maybe something for a follow-up refactor).

Either way, please add a doc comment to this type to explain why it exists.